scrapy/downloadermiddlewares/httpproxy.py

```diff
@@ -9,11 +9,13 @@
 
 from scrapy.utils.httpobj import urlparse_cached
 from scrapy.exceptions import NotConfigured
+from scrapy.utils.python import to_bytes
 
 
 class HttpProxyMiddleware(object):
 
-    def __init__(self):
+    def __init__(self, auth_encoding=None):
```

Comment:
Why is this argument optional? Passing None means a default of utf-8, I think it is not good to have 2 different default encodings.